examples/applications/restapi/handlers.py

```diff
@@ -302,35 +302,55 @@ async def create_tx(self, request):
         - Should handle any kind of output script.( see bitcoinx.address for
         utilities for building p2pkh, multisig etc outputs as hex strings.)
         """
+        frozen_utxos = []
+        account = None
+        tx = None
         try:
             tx, account, password = await self._create_tx_helper(request)
             self.raise_for_duplicate_tx(tx)
             account.sign_transaction(tx, password)
 
-            _frozen_utxos = self.app_state.app.get_and_set_frozen_utxos_for_tx(tx, account)
+            frozen_utxos = self.app_state.app.get_and_set_frozen_utxos_for_tx(tx, account)
             response = {"value": {"txid": tx.txid(),
                                   "rawtx": str(tx)}}
             return good_response(response)
         except Fault as e:
+            if tx and tx.is_complete() and e.code != Fault(Errors.ALREADY_SENT_TRANSACTION_CODE):
+                self.cleanup_tx(tx, account, frozen_utxos)
             return fault_to_http_response(e)
+        except Exception as e:
+            if tx and tx.is_complete():
+                self.cleanup_tx(tx, account, None)
+            return fault_to_http_response(
+                Fault(code=Errors.GENERIC_INTERNAL_SERVER_ERROR, message=str(e)))
 
     async def create_and_broadcast(self, request):
+        frozen_utxos = []
+        account = None
+        tx = None
         try:
             tx, account, password = await self._create_tx_helper(request)
             self.raise_for_duplicate_tx(tx)
             account.sign_transaction(tx, password)
             frozen_utxos = self.app_state.app.get_and_set_frozen_utxos_for_tx(tx, account)
-            result = await self._broadcast_transaction(str(tx), tx.hash(), account)
+            try:
+                result = await self._broadcast_transaction(str(tx), tx.hash(), account)
+            except aiorpcx.jsonrpc.RPCError as e:
+                raise Fault(Errors.AIORPCX_ERROR_CODE, e.message)
             self.prev_transaction = result
             response = {"value": {"txid": result}}
             self.logger.debug("successful broadcast for %s", result)
             return good_response(response)
         except Fault as e:
+            if len(frozen_utxos) != 0 and e.code != Errors.ALREADY_SENT_TRANSACTION_CODE:
+                self.cleanup_tx(tx, account, frozen_utxos)
             return fault_to_http_response(e)
-        except aiorpcx.jsonrpc.RPCError as e:
-            account.set_frozen_coin_state(frozen_utxos, False)
-            self.remove_signed_transaction(tx, account)
-            return fault_to_http_response(Fault(Errors.AIORPCX_ERROR_CODE, e.message))
+        except Exception as e:
+            self.logger.exception(e)
```

Comment:
I am not sure this is valid. `.exception` should pick the exception from the context, and I think it expects the first argument to be a message.